utils/get_prometheus_data.py

```diff
@@ -0,0 +1,145 @@
+import os
+import requests
+import json
+import logging
+import urllib3
+from datetime import datetime, timedelta
+import time
+import sys
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+logger = logging.getLogger("snafu")
+
+class get_prometheus_data():
+    def __init__(self, action):
+
+        self.uuid = action["uuid"]
+        self.user = action["user"]
+        self.cluster_name = action["cluster_name"]
+        self.test_config = action["test_config"]
+        self.start = action["starttime"]
+
+        # change datetime in seconds string to datetime object
+        endtime_object = datetime.fromtimestamp(int(action["endtime"]))
+        # add 120s buffer to end time
+        endtime_object = endtime_object + timedelta(seconds=120)
+        # convert back to string
+        self.end = endtime_object.strftime('%s')
+
+        # step value to be used in prometheus query
+        self.T_Delta = 30
+
+        self.get_data = False
+        if "snafu_token" in os.environ and "prometheus_url" in os.environ:
+            self.get_data = True
+            token = os.environ["snafu_token"]
+            self.url = os.environ["prometheus_url"]
+            bearer = "Bearer " + token
+            self.headers = {'Authorization': bearer}
+        else:
+            logger.warn("""snafu service account token and prometheus url not set \n
+                        No Prometheus data will be indexed""")
+
+    def api_call(self, query):
```

Comment:
can we look at reusing https://prometheus-api-client-python.readthedocs.io/source/prometheus_api_client.html library here ?

Reply:
If this is desired it will require a rewrite of most of get_prometheus_data.py, also not sure if it will function the same.

Your call @aakarshg @jtaleric.

Reply:
this will offload a lot of burden of your head to keep maintaining this interface to connect/retrieve metrics. @amitsagtani97 also has a pr in place to get aggregations (https://github.com/AICoE/prometheus-api-client-python/pull/99 )

Reply:
How would you go about adding this python module to the container?

Reply:
we can add this to requirements.txt and then it'll get auto pulled when we do an install of snafu, otherwise we add it to dockerfile to also do a pip install of the module as follows:
```
pip install prometheus-api-client
```